tests/tests.rs

```diff
@@ -1470,3 +1470,186 @@ fn test_base_directory() {
         ),
     );
 }
+
+/// --prune
+#[test]
+fn test_prune() {
+    let dirs = &["foo", "bar/zap/foo"];
+    let files = &[
+        "foo/a.foo",
+        "foo/b.bar",
+        "bar/zap/c.foo",
+        "bar/zap/foo/d.foo",
+    ];
+    let te = TestEnv::new(dirs, files);
+
+    // without pruning
+    te.assert_output(
+        &["foo"],
+        "bar/zap/foo/d.foo
+         bar/zap/c.foo
+         bar/zap/foo
+         foo/a.foo
+         foo",
+    );
+    // with pruning
+    te.assert_output(
+        &["--prune", "foo"],
+        "bar/zap/c.foo
+         bar/zap/foo
+         foo",
+    );
+
+    // with --full-path
+    let root = te.system_root();
+    let prefix = escape(&root.to_string_lossy());
+    // before prune
+    te.assert_output(
+        &[
+            "--full-path",
+            &format!("^{prefix}.*zap.*foo$", prefix = prefix),
+        ],
+        "bar/zap/c.foo
+         bar/zap/foo/d.foo
+         bar/zap/foo",
+    );
+    // after prune
+    te.assert_output(
+        &[
+            "--prune",
+            "--full-path",
+            &format!("^{prefix}.*zap.*foo$", prefix = prefix),
+        ],
+        "bar/zap/c.foo
+         bar/zap/foo",
+    );
+
+    // --type f
+    te.assert_output(&["--prune", "--type", "f", "foo"], "bar/zap/c.foo");
+    // --type d
+    te.assert_output(
+        &["--prune", "--type", "d", "foo"],
+        "bar/zap/foo
+        foo",
+    );
+    // --prune with other options but no pattern
+    te.assert_output(
+        &["--prune", "--type", "d"],
+        "bar/zap/foo
+         bar/zap
+         bar
+         foo",
+    );
+
+    // --extention
+    // before prune
+    te.assert_output(&["--extension", "bar"], "foo/b.bar");
+    // after prune
+    te.assert_output(&["--prune", "--extension", "bar", "foo"], "");
+
+    // --max-depth
+    // before prune
+    te.assert_output(
+        &["--max-depth", "3", "foo"],
+        "bar/zap/foo
+         bar/zap/c.foo
+         foo/a.foo
+         foo",
+    );
+    // after prune
+    te.assert_output(
+        &["--prune", "--max-depth", "3", "foo"],
+        "bar/zap/foo
+         bar/zap/c.foo
+         foo",
+    );
+
+    // --exclude
+    // before prune
+    te.assert_output(
+        &["foo", "--exclude", "zap"],
+        "foo
+         foo/a.foo",
+    );
+    // after prune
+    te.assert_output(&["foo", "--prune", "--exclude", "zap"], "foo");
+
+    // --follow
+    // TODO
+    //te.assert_output(
+    //    &["--follow", "c.foo"],
+    //    "bar/zap/c.foo
+    //    symlink/c.foo",
+    //);
+
+    // --size
```

Comment:
same here: size would be similar to `--type`

Reply:
removed in commit 3